CI: Sign test built APPX package with self-signed key

starnight opened the pull request:
To make testers test easily, sign the test APPX package with self-signed
key. Then, testers can installed the signed APPX package directly.

The Publisher in appxmanifest.xml must be the same with the
certificate's subject. And, the selfsign.pfx must match secrets'
SELFSIGN_PHRASE.

https://phabricator.endlessm.com/T33481

starnight committed:
CI: Sign built package with self-signed key for test

To make testers test easily, sign the test APPX package with self-signed
key. Then, testers can installed the signed APPX package directly.

The Publisher in appxmanifest.xml must be the same with the
certificate's subject. And, the selfsign.pfx must match secrets'
SELFSIGN_PHRASE.

https://phabricator.endlessm.com/T33481

jprvita commented on .github/workflows/main.yml:
```diff
@@ -129,29 +123,27 @@ jobs:
+          name: "endlesskeyapp.testsigned.appx"
+          path: "endlesskeyapp.testsigned.appx"
+
+      - name: Sign artifact for release
```
We don't need to sign this for release -- it looks like the Microsoft Store accepts an unsigned package, so there is no need for this step anymore.

starnight replied:
ok

jprvita commented on .github/workflows/main.yml:
```diff
@@ -129,29 +123,27 @@ jobs:
+          name: "endlesskeyapp.testsigned.appx"
+          path: "endlesskeyapp.testsigned.appx"
+
+      - name: Sign artifact for release
```
We don't need to sign this for release -- it looks like the Microsoft Store accepts an unsigned package, so there is no need for this step anymore.

jprvita replied:
Sorry, just to be clear -- we are planning to use SignPath in the future to sign the .exe for the Endless Key USB, but that should be handled in a separate ticket, as we need to define a new signing policy and artifact configuration on their platform first.

starnight replied:
ok

jprvita commented on .github/workflows/main.yml:
```diff
@@ -163,14 +155,14 @@ jobs:
 
-      - name: Upload signed APPX
-        if: steps.sign_for_release.outcome == 'success' || steps.sign_for_testing.outcome == 'success'
+      - name: Upload signed APPX for release
```
Here as well, we don't need to sign anything during a release, so this step is not needed anymore.

starnight replied:
ok

jprvita commented on .github/workflows/main.yml:
```diff
@@ -163,14 +155,14 @@ jobs:
 
-      - name: Upload signed APPX
-        if: steps.sign_for_release.outcome == 'success' || steps.sign_for_testing.outcome == 'success'
+      - name: Upload signed APPX for release
```
Here as well, we don't need to sign anything during a release, so this step is not needed anymore.

starnight replied:
ok

jprvita commented on .github/workflows/main.yml:
```diff
@@ -129,29 +123,27 @@ jobs:
-            -WaitForCompletion
+          & "${Env:ProgramFiles(x86)}\Windows Kits\10\App Certification Kit\signtool.exe" `
+            sign /fd SHA256 /a /f .buildkite_win/selfsign.pfx /p $env:SIGN_PHRASE endlesskeyapp.unsigned.appx
+          mv endlesskeyapp.unsigned.appx endlesskeyapp.testsigned.appx
```
Let's drop the `.unsigned` part from the unsigned package (on the "Packaging" step a few lines above), since this is what we will release in the end (an unsigned package). Then maybe just call `.selfsigned` here instead of `.testsigned`?

Also, I think we should drop `app` from the name and just call it `endlesskey.appx`, but let's leave that part for a separate commit.

starnight replied:
ok

jprvita commented on .github/workflows/main.yml:
```diff
@@ -129,29 +123,27 @@ jobs:
-            -WaitForCompletion
+          & "${Env:ProgramFiles(x86)}\Windows Kits\10\App Certification Kit\signtool.exe" `
+            sign /fd SHA256 /a /f .buildkite_win/selfsign.pfx /p $env:SIGN_PHRASE endlesskeyapp.unsigned.appx
+          mv endlesskeyapp.unsigned.appx endlesskeyapp.testsigned.appx
```
Let's drop the `.unsigned` part from the unsigned package (on the "Packaging" step a few lines above), since this is what we will release in the end (an unsigned package). Then maybe just call `.selfsigned` here instead of `.testsigned`?

Also, I think we should drop `app` from the name and just call it `endlesskey.appx`, but let's leave that part for a separate commit.

starnight replied:
ok

starnight committed:
CI: Release unsigned package for publishing to Microsoft store

According to the study [1], we learned Microsoft will verify and sign
the packages on the store. So, release the unsigned package to publish
to Microsoft store.

[1] https://phabricator.endlessm.com/T33456

jprvita commented on .github/workflows/main.yml:
```diff
@@ -121,63 +115,35 @@ jobs:
           cp .buildkite_win/appxmanifest.xml kolibri-windows/
-          & "${Env:ProgramFiles(x86)}\Windows Kits\10\App Certification Kit\makeappx.exe" pack /d kolibri-windows /p endlesskeyapp.unsigned.appx
-
-      - name: Upload unsigned APPX
```
Let's keep this stage uploading the unsigned package for all builds, so we can easily get it for checking if it pass validation with the store.

starnight replied:
ok

jprvita commented on .github/workflows/main.yml:
```diff
@@ -121,63 +115,35 @@ jobs:
+          name: "endlesskey.selfsigned.appx"
+          path: "endlesskey.selfsigned.appx"
+
+      - name: Upload unsigned package for release
```
And since we will upload the unsigned package on every build, before signing, we don't need this stage on releases (only the next "Release" stage to create the release on GitHub).

starnight replied:
ok

starnight committed:
CI: Align the package name with endlesskey

The DisplayName in appxmanifest.xml has been defined as "Endless Key" in
commit 7cd6f30e43ac ("appx: Remove 'app' from name").

https://phabricator.endlessm.com/T33456
